Backend/BackendAPI/Startup.cs

```diff
@@ -27,10 +29,13 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddDbContext<DataContext>(options =>
+            services.AddScoped<IBikeRepository, BikeRepository>();
+
+            services.AddDbContextPool<DataContext>(options =>
```

Comment:
Przyznam się że poczytałem ale i tak nie do końca rozumiem różnicy miedzy AddDbContext, ale zakładam że to ejst przemyślane (chętnie przy okazji dowiem się z czego wynika ta zmiana)

Reply:
Róznica wynika z tworzenia nowych połączeń albo korzystania z puli połączeń do bazy (zagadnienie bylo omawiane na wykladzie o jdbc na Bazach danych na 2 roku; nazywalo sie to connection pool wlasnie)